services/aur_cache/main.go

```diff
@@ -4,39 +4,85 @@ import (
 	"encoding/json"
 	"fmt"
 	"github.com/finitum/AAAAA/pkg/aur"
+	"github.com/finitum/AAAAA/pkg/store"
 	"github.com/go-chi/chi"
 	"github.com/go-chi/chi/middleware"
 	"github.com/go-chi/cors"
 	"github.com/go-chi/render"
 	log "github.com/sirupsen/logrus"
 	"net/http"
+	"os"
+	"strings"
 )
 
-func main()  {
+func main() {
 	r := chi.NewRouter()
 	r.Use(middleware.StripSlashes)
 	r.Use(middleware.Logger)
 	r.Use(middleware.Recoverer)
 
 	r.Use(cors.Handler(cors.Options{AllowedOrigins: []string{"*"}}))
-
 	r.Use(render.SetContentType(render.ContentTypeJSON))
 
-	r.Get("/search/{term}", proxy)
+	cache, err := store.OpenBadgerCache(os.TempDir() + "/AAAAA")
+	if err != nil {
+		log.Fatalf("Couldn't open store (%v)", err)
+	}
+
+	r.Get("/search/{term}", func(writer http.ResponseWriter, request *http.Request) {
```

Comment:
If you make proxy return a handler this can be made cleaner

Reply:
what do you mean?